v1/citoid.yaml

```diff
@@ -0,0 +1,111 @@
+swagger: '2.0'
+info:
+  version: '1.0.0-beta'
+  title: MediaWiki Citation API
+  description: Citation API based on the Citoid services
+  termsOfService: https://github.com/wikimedia/restbase#restbase
+  contact:
+    name: Services
+    email: services@lists.wikimedia.org
+    url: https://www.mediawiki.org/wiki/Services
+  license:
+    name: Apache licence, v2
+    url: https://www.apache.org/licenses/LICENSE-2.0
+paths:
+  /citation/{format}/{query}:
+    get:
+      tags:
+        - Citation
+      summary: Get citation data given an article identifyer.
+      description: |
+        Generates citation data given a URL, DOI, PMID, or PMCID.
+
+        See more at [Citoid service documentation](https://www.mediawiki.org/wiki/Citoid)
+
+        Stability: [experimental](https://www.mediawiki.org/wiki/API_versioning#Experimental)
+      produces:
+        - application/json; charset=utf-8;
+        - application/x-bibtex; charset=utf-8
+      parameters:
+        - name: format
+          in: path
+          description: "The format to use for the resulting citation data"
+          enum:
+            - mediawiki
+            - bibtex
+            - zotero
+          required: true
+        - name: query
+          in: path
+          description: >
+            URL of an article, DOI, PMCID or PMID in the URL-encoded format.
+            Note that on the Swagger-UI doc page you don't need to URI-encode the
+            parameter manually, it will be done by the docs engine.
+          type: string
+          required: true
+        - name: basefields
+          in: query
+          type: boolean
+          required: false
+          description: >
+            Optional boolean argument. If it is set to true (or 1), it will return the base
+            fields for the data in addition to the mapped fields. For example, it will return
+            `publicationTitle` in addition to `bookTitle` or `websiteTitle`
+        - name: Accept-Language
+          in: header
+          type: string
+          required: false
+          description: >
+            For some articles the result depends on the `Accept-Language` header, so
+            provide it if localized content is required.
+      responses:
+        '200':
+          description: The citation data in the requested format
+        '404':
+          description: Citation data was not found.
+          schema:
+            $ref: '#/definitions/problem'
+        default:
+          description: Error
+          schema:
+            $ref: '#/definitions/problem'
+
+      x-request-handler:
+        - get_citation_from_backend:
+            request:
+              method: get
+              uri: '{{options.host}}/api?format={format}&search={query}'
+              headers:
+                accept-language: '{{accept-language}}'
```

Comment:
@berndsi That would be an internal RESTBase-only endpoint, we have no intension to expose it publicly or to other services (right now, we may change the opinion if that feels useful)

Why is that more hacky is because normally client would set the `accept-language` header, so the information obtained from that endpoint would be useful only on a small percentage of requests, but our current tech powering the request templates would require us to fetch it regardless of whether the `accept-language` was provided or not.